lib/repo_tools.pm

```diff
@@ -35,6 +36,41 @@ our @EXPORT = qw(
   disable_oss_repo
   generate_version);
 
+=head2 add_qa_head_repo
+
+    add_qa_head_repo([url_only => 1]);
+
+Helper to add QA:HEAD repository repository.
+Use default values if QA_HEAD_REPO is not set.
+Suitable for both SLES and openSUSE.
+=cut
+sub add_qa_head_repo {
+    my %args     = @_;
+    my $url_only = $args{url_only};
+    my $repo     = get_var('QA_HEAD_REPO');
+    my $version;
+    if (!$repo) {
+        my $distro       = get_distro();
+        my $version      = get_version();
+        my $prod_version = get_product_version();
+        if (is_opensuse) {
+            $repo = 'https://download.opensuse.org/repositories/devel:/openSUSE:/QA:/';
+            $repo .= "$distro:/$prod_version/openSUSE_${distro}_$version/";
+        } elsif (is_sle) {
+            $repo = "http://download.suse.de/ibs/QA:/Head/SLE-$version/";
+        }
+        die 'QA_HEAD_REPO not detected' unless ($repo);
```

Comment:
I personally don't see a benefit to auto-generate repository:
* QA_HEAD_REPO is usually defined for SLES on medium level for all jobs, if it is missing, it is incorrect configuration at medium level
*  devel:openSUSE:QA on openSUSE is something which i snot used much, if at all, it contains only one package for systemd test suite, failing on Tumbleweed and for Leap 15.1 repo doesn't exist

Reply:
But now I see, that code was already there and you just moved it o lib.

Reply:
@SergioAtSUSE Could you please have a look as well? As you introduced this part into systemd test suite.

Reply:
@czerw If `QA_HEAD_REPO` is on all required jobs, we indeed don't have to generate it.
Maybe it's not worth of having a function doing just simple:

```
zypper_ar(get_required_var('QA_HEAD_REPO'), 'qa_repo');
zypper_call('--gpg-auto-import-keys ref');
```
but I'd still be for adding it. Having a single function in library which adds repository (there are several versions of this functionality in tests, it's worth of cleanup it). Or at least cleanup the detection. Duplicity in tests is enormous :).

Reply:
@pevik sounds good

Reply:
Taking a look.
The problem using the SETTING as default is that we have to hardcode the link and we cannot generate it.
So, the way it should work is having the URL generated in the test for each distribution/version.
But, I added the QA_HEAD_REPO check to allow tsaupe and me to override the repository with our home branched and test changes without touching the code.